client/components/filter-select-control/index.tsx

```diff
@@ -0,0 +1,250 @@
+/**
+ * This is a copy of Gutenberg's CustomSelectControl component, found here:
+ * https://github.com/WordPress/gutenberg/tree/7aa042605ff42bb437e650c39132c0aa8eb4ef95/packages/components/src/custom-select-control
+ *
+ * It has been forked from the existing WooPayments copy of this component (client/components/custom-select-control)
+ * to match this specific select input design with an inline label and option hints.
```

Comment:
Would be good to add to this comment a rough plan for how we might merge this upstream or reduce number of forks in WooPayments code base 😁

Reply:
Update: I've paused my review because I think I need to review the diff from the Gutenberg code (and maybe other WooPayments `CustomSelectControl` forks. @Jinksi  how do you think we should handle this?